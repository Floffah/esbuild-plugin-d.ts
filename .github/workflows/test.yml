name: Test

on: [push, pull_request]

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: oven-sh/setup-bun@v1
      - run: bun install --frozen-lockfile
      - run: bun run test
