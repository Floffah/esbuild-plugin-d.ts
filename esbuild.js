const {build} = require("esbuild");

const dev = process.argv.includes("--dev") || process.argv.includes("-d");

build({
    bundle: true,
    target: "node16.0",
    platform: "node",
    watch: dev,
    color: true,
    logLevel: "info",
    minify: !dev,
    minifyIdentifiers: !dev,
    minifySyntax: !dev,
    minifyWhitespace: !dev,
    format: "cjs",
    entryPoints: ["./src/index.ts"],
    outfile: "./dist/index.cjs",
    external: ["esbuild", "typescript", "chalk"]
})
