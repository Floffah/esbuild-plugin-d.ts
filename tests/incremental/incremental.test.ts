import { dtsPlugin } from "../../dist";
import { clearDistDir, distDir, readOutputFile } from "../_utils";
import { expect, test } from "bun:test";
import { build } from "esbuild";
import { existsSync } from "fs";
import { resolve } from "path";

test("Incremental mode", async () => {
    const tsconfig = resolve(__dirname, "./tsconfig.json");
    const __buildContext = Date.now();

    await build({
        plugins: [dtsPlugin({ tsconfig, __buildContext })],
        entryPoints: [resolve(__dirname, "./incremental.ts")],
        outdir: distDir,
        tsconfig,
    });

    expect(readOutputFile("incremental")).toMatchSnapshot();

    clearDistDir();

    await build({
        plugins: [dtsPlugin({ tsconfig, __buildContext })],
        entryPoints: [resolve(__dirname, "./incremental.ts")],
        outdir: distDir,
        tsconfig,
    });

    expect(existsSync(resolve(distDir, "incremental.d.ts"))).toBe(false);
});
