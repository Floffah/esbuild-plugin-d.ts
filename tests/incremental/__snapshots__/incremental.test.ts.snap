// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Incremental mode 1`] = `
"export declare function someFunction(a: string, b: number): string;
export type SomeType = "someValue" | "anotherValue";
export declare const someVar: SomeType;
"
`;
